src/features/filter.ts

```diff
@@ -1,10 +1,59 @@
-export const actions = { /* put action creators here */};
+// export type Status = 'all' | 'active' | 'completed';
+type Filters = {
+  query: string,
+  status: string,
+};
+
+type StatusAction = {
+  type: 'filters/Status'
+  query: string,
+  status: string
+};
 
-const filterReducer = () => {
-  return {
-    query: '',
-    status: 'all',
-  };
+type QueryAction = {
+  type: 'filters/Query'
+  query: string,
+  status: string
 };
 
+const editQuery = (filters: Filters, value: string): QueryAction => ({
+  ...filters,
+  type: 'filters/Query',
+  query: value,
+});
+
+const editStatus = (filters: Filters, value: string): StatusAction => ({
+  ...filters,
+  type: 'filters/Status',
+  status: value,
+});
+
+type Actions = StatusAction | QueryAction;
+
+const startSilters: Filters = {
+  query: '',
+  status: 'all',
+};
+
+const filterReducer = (filters: Filters = startSilters, actions: Actions) => {
```

Comment:
In my opinion, you don't need to create a separate store for filters, it can be part of todosStore, because if more entities appear in the app that will also have filtering, it can be a problem